File: app/src/main/java/aki/pvnghe/unittestapi/service/GithubUserService.java
package aki.pvnghe.unittestapi.service;


import aki.pvnghe.unittestapi.model.User;
import aki.pvnghe.unittestapi.model.UsersList;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;
import rx.Observable;

public interface GithubUserService {

    @GET("/search/users?per_page=2")
    Observable<UsersList> searchGithubUsers(@Query("q") String searchTerm);

    @GET("/users/{username}")
    Observable<User> getUser(@Path("username") String username);
}

File: data/src/main/java/aki/pvnghe/data/service/GithubUserService.kt
package aki.pvnghe.data.service

import aki.pvnghe.data.model.User
import aki.pvnghe.data.model.UsersList
import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface GithubUserService {
    @GET("/search/users?per_page=10")
    fun searchGithubUsers(@Query("q") searchTerm: String?): Single<UsersList>

    @GET("/users/{username}")
    fun getUser(@Path("username") username: String?): Single<User>
}